eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -281,11 +284,15 @@ class ReplicationEndpoint(
       val promise = Promise[Unit]()
       val recovery = new Recovery(this)
 
+      def recoveryFailure[U](partialUpdate: Boolean): PartialFunction[Throwable, Future[U]] = {
+        case t => Future.failed(new RecoveryException(t, partialUpdate))
+      }
+
       val recoveryOperation = for {
-        infos <- recovery.readEndpointInfos
-        clocks <- recovery.readEventLogClocks
+        infos <- recovery.readEndpointInfos.recoverWith(recoveryFailure(partialUpdate = false))
+        clocks <- recovery.readEventLogClocks.recoverWith(recoveryFailure(partialUpdate = false))
         links = recovery.recoveryLinks(infos, clocks)
-        _ <- recovery.deleteSnapshots(links)
+        _ <- recovery.deleteSnapshots(links).recoverWith(recoveryFailure(partialUpdate = true))
```

Comment:
Just to be sure: Is `partialUpdate = true` here because the application might have deleted events under the assumption that the state is stored in the snapshot?